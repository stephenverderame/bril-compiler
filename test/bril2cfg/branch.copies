digraph {
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[]\n----------\nx := 42\ny := 30\ny2 := 5\nz := y - y2\nb := x < z\nif b\n----------\n[]", shape="rectangle", style="rounded"];
		main_3 [label="[]\n----------\nprint x\n----------\n[]", shape="rectangle", style="rounded"];
		main_4 [label="[]\n----------\nprint y\n----------\n[]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3 [label="T"];
		main_2 -> main_4 [label="F"];
	}
}
