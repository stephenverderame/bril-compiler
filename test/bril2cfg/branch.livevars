digraph {
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[]\n----------\nx := 42\ny := 30\ny2 := 5\nz := y - y2\nb := x < z\nif b\n----------\n[x, y]", shape="rectangle", style="rounded"];
		main_3 [label="[x]\n----------\nprint x\n----------\n[]", shape="rectangle", style="rounded"];
		main_4 [label="[y]\n----------\nprint y\n----------\n[]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3 [label="T"];
		main_2 -> main_4 [label="F"];
	}
}
