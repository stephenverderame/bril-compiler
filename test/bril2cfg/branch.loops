digraph {
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="x := 42\ny := 30\ny2 := 5\nz := y - y2\nb := x < z\nif b", shape="rectangle", style="rounded"];
		main_3 [label="print x", shape="rectangle", style="rounded"];
		main_4 [label="print y", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3 [label="T"];
		main_2 -> main_4 [label="F"];
	}
}
