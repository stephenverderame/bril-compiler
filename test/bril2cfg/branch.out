digraph {
	subgraph cluster_main {
		label="main";
		rankdir="TB";
		0 [label="START", shape="ellipse"];
		2 [label="x := 42", shape="ellipse"];
		3 [label="y := 30", shape="ellipse"];
		4 [label="y2 := 5", shape="ellipse"];
		5 [label="z := y - y2", shape="ellipse"];
		6 [label="b := x < z", shape="ellipse"];
		7 [label="if b", shape="ellipse"];
		8 [label="print x", shape="ellipse"];
		10 [label="print y", shape="ellipse"];

		0 -> 2;
		2 -> 3;
		3 -> 4;
		4 -> 5;
		5 -> 6;
		6 -> 7;
		7 -> 8 [label="T"];
		7 -> 10 [label="F"];
	}
}
